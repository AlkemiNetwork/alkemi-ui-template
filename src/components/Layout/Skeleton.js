import React from "react";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";

import LayoutContent from "../../containers/Layout/Content";
import Sidebar from "../../containers/Sidebar/Sidebar";
import AppBar from "../../containers/AppBar/AppBar";

const styles = theme => ({
    frame: {
        position: "relative",
        display: "flex",
        width: "100%",
        minHeight: "100%"
    },
    content: {
        width: "100%",
        flexGrow: 1,
        backgroundColor: theme.palette.background.default,
        minHeight: "calc(100% - 56px)",
        marginTop: 56,
        [theme.breakpoints.up("sm")]: {
            minHeight: "calc(100% - 64px)",
            marginTop: 64
        }
    }
});

const LayoutSkeleton = ({ classes }) => (
    <div className={classes.frame}>
        <AppBar />
        <Sidebar />
        <div className={classes.content}>
            <LayoutContent />
        </div>
    </div>
);

LayoutSkeleton.propTypes = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(LayoutSkeleton);
