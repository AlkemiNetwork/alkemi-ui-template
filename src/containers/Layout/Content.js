import React from "react";
import { Route, Switch } from "react-router";

import Home from "../../components/Home/Home";
import NoWhere from "../../components/NoWhere/NoWhere";
import { HOME } from "../../constants/routes";

const LayoutContent = () => (
    <Switch>
        <Route path={HOME} exact component={Home} />
        <Route component={NoWhere} />
    </Switch>
);

export default LayoutContent;
