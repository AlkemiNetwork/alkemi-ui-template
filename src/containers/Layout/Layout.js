import React from "react";
import { Switch, Route, Redirect } from "react-router";

import LayoutSkeleton from "../../components/Layout/Skeleton";
import { HOME } from "../../constants/routes";

const Layout = () => (
    <Switch>
        <Route path="/" exact component={() => <Redirect to={HOME} />} />
        <Route component={LayoutSkeleton} />
    </Switch>
);

export default Layout;
