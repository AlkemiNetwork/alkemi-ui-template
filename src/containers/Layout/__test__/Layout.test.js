import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Route, Switch } from "react-router";

import Layout from "../Layout";

describe("<Layout />", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Layout />);
    });

    it("renders 1 <Switch /> components", () => {
        expect(wrapper.find(Switch).length).toEqual(1);
    });

    it("renders 2 <Route /> components", () => {
        expect(wrapper.find(Route).length).toEqual(2);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
