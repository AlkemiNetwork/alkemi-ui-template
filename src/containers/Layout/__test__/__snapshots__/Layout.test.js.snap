// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> matches snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
